cpp/include/replace.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2018-2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef REPLACE_HPP
+#define REPLACE_HPP
+
+#include <cudf.h>
+#include <types.hpp>
+
+// Forward declaration
+typedef struct CUstream_st* cudaStream_t;
+
+namespace cudf {
+
+/**
+ * @brief Replaces all null values in a column with corresponding values of another column
+ *
+ * The first column is expected to be a regular gdf_column. The second column
+ * must be of the same type and same size as the first.
+ *
+ * The function replaces all nulls of the first column with the
```

Comment:
```suggestion
 * Performs the operation `output[i] = (input[i] == NULL) ? replacement[i] : input[i]`
```

Reply:
Isn't this harder to read? If the docs aren't clear enough, perhaps an example would help more than the algorithm?

Reply:
I think there's a little room to make it more concise, but also just trying to clarify what "corresponding" element means.

Reply:
Agreed. Here's my proposal, and if you don't like it, we'll go with your suggested change:

> For each index in the input column where it is null, replaces it with the value in that index of the replacement column.

Reply:
> For each index in the input column where it is null, replaces it with the value in that index of the replacement column.

That's still a bit unclear because we're making a new column, not doing the operation in place. 

> If `input[i]` is valid, its value will be copied to `output[i]`, else `replacements[i]` will be copied